subsys/shell/shell.c

```diff
@@ -1070,8 +967,15 @@ static int shell_execute(const struct shell *shell)
 			/* Command called with help option so it makes no sense
 			 * to search deeper commands.
 			 */
-			help_flag_set(shell);
-			break;
+			if (help_entry.help) {
+				shell->ctx->active_cmd = help_entry;
+				shell_help_print(shell);
+				return 1;
+			}
+
+			shell_fprintf(shell, SHELL_ERROR,
```

Comment:
you could use `shell_error()` here

Reply:
I decided to keep `shell_printf` in shell.c